File: android-app/baselibrary/src/main/java/com/musketeer/baselibrary/BaseApplication.java
package com.musketeer.baselibrary;

import android.app.Application;
import android.content.Intent;

/**
 * Created by zhongxuqi on 15-10-24.
 */
public class BaseApplication extends Application {
    public static BaseApplication instance;

    public static BaseApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public void startService(Class<?> cls) {
        Intent intent=new Intent();
        intent.setClass(this, cls);
        startService(intent);
    }
}

File: android-app/app/src/main/kotlin/com/musketeer/scratchpaper/BaseApplication.kt
package com.musketeer.scratchpaper

import android.app.Application
import android.content.Intent

/**
 * Created by zhongxuqi on 15-10-24.
 */
open class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    fun startService(cls: Class<*>) {
        val intent = Intent()
        intent.setClass(this, cls)
        startService(intent)
    }

    companion object {
        var instance: BaseApplication? = null
    }
}
